### PR TITLE
Add folder name to test weaver

### PR DESCRIPTION
**Pull request by GeertvanHorrik**
See #1297

### CODE REVIEW

File: FodyHelpers/Testing/WeaverTestHelper.cs [hunk 1]
```diff
@@ -12,13 +12,20 @@ public static TestResult ExecuteTestRun(
         bool runPeVerify = true,
         Action<ModuleDefinition>? afterExecuteCallback = null,
         Action<ModuleDefinition>? beforeExecuteCallback = null,
+        string? folderName = null,
```
Comment (tom-englert) [inline FodyHelpers/Testing/WeaverTestHelper.cs:15]: If you move this new parameter to the end, it should be fully backwards compatible

Comment (GeertvanHorrik) [reply FodyHelpers/Testing/WeaverTestHelper.cs:15]: I have though about it. You mean behind the assemblyName or really at the end? On the long term, I think it's better to keep them together, but short-term would be best to add at the very end.

Comment (ltrzesniewski) [reply FodyHelpers/Testing/WeaverTestHelper.cs:15]: I agree the `folderName` parameter should be put at the very end, this avoids a breaking change.

**Review by ltrzesniewski — COMMENTED** [review]
I had a very similar issue in InlineIL, and I had to basically [rewrite `WeaverTestHelper`](https://github.com/ltrzesniewski/InlineIL.Fody/blob/50ec036db400feeb5a991d9a328d22efffb7e7f5/src/InlineIL.Tests/Weaving/WeaverRunner.cs), so I completely understand the need for this change. 🙂

**Comment by GeertvanHorrik** [comment]
Updated the PR based on feedback. Thanks!

**Review by tom-englert — APPROVED** [review]

**Review by ltrzesniewski — APPROVED** [review]

**Pull request merged**